### DISCUSSION TITLE
SQL `SHUTDOWN` Command or Endpoint?

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#694] [Ideas]
Dear Developers,

as far as I can see there is no way to remotely shutdown the database server, apart from SSHing into the machine, correct?

If this is the case, I would like to propose either a SQL `SHUTDOWN` command or alternatively an `/api/v1/shutdown` endpoint. I think it would be useful to have such a command, like for example [Redis](https://redis.io/commands/shutdown/) has, so one can safely shutdown a server. However, maybe such a command needs to be limited to the `root` user?

Thank you for considering this proposal

PS: See also: https://github.com/ArcadeData/arcadedb/discussions/641

**Comment by gramian** [comment]
I guess to better limit this command to the root user(s), a shutdown interface should rather be a (POST) endpoint than a (SQL) command.

**Comment by gramian** [comment]
One argument for a `SHUTDOWN` command is that database lock files could be cleaned up.

**Comment by lvca** [comment]
I like it, I've created a new issue from this discussion: https://github.com/ArcadeData/arcadedb/issues/722.
